libsplinter/src/rest_api/actix_web_3/api.rs

```diff
@@ -0,0 +1,176 @@
+// Copyright 2018-2021 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Contains the implementation of `RestApi`.
+
+use std::future::Future;
+use std::io::Error as IoError;
+use std::net::SocketAddr;
+use std::pin::Pin;
+use std::sync::{mpsc, Arc, Mutex};
+use std::thread::{self, JoinHandle};
+use std::time::Duration;
+
+use actix_0_10::System as ActixSystem;
+use actix_web_3::{dev::Server, middleware, App, HttpServer};
+use futures_0_3::executor::block_on;
+use openssl::ssl::SslAcceptorBuilder;
+
+use crate::error::InternalError;
+use crate::rest_api::RestApiServerError;
+use crate::threading::shutdown::ShutdownHandle;
+
+use super::ResourceProvider;
+
+/// Contains information about the ports to which the REST API is bound.
+#[derive(Debug)]
+pub struct BindAddress {
+    /// The SocketAddr which defines the bound port.
+    pub addr: SocketAddr,
+
+    /// The scheme (such as http) that is running on this port.
+    pub scheme: String,
+}
+
+enum FromThreadMessage {
```

Comment:
Could this enum just be a `Result`?

Reply:
Technically. However, I don't mean this as a Result in meaning, but as the channel API definition. Using enum for things sent over channels lets you name it--FromThreadMessage--and leaves room for sending additional messages over the channel in the future without changing the fundamental channel type. (This is a general reason, not specific to this situation.)

Reply:
This and the loop{} for the channel will be considered prior to feature stabilization.